src/block_cache/block_chunkcache.c

```diff
@@ -739,6 +780,55 @@ __wt_chunkcache_remove(
     return (0);
 }
 
+/*
+ * __wt_chunkcache_ingest --
+ *     Read all the contents from a file and insert it into the chunkcache.
+ */
+int
+__wt_chunkcache_ingest(
+  WT_SESSION_IMPL *session, const char *local_name, const char *sp_obj_name, uint32_t objectid)
+{
+    WT_CHUNKCACHE *chunkcache;
+    WT_CHUNKCACHE_CHUNK *chunk;
+    WT_CHUNKCACHE_HASHID hash_id;
+    WT_DECL_RET;
+    WT_FH *fh;
+    wt_off_t already_read, size;
+    uint64_t bucket_id;
+
+    chunkcache = &S2C(session)->chunkcache;
+    already_read = 0;
+
+    if (!F_ISSET(chunkcache, WT_CHUNKCACHE_CONFIGURED) ||
+      !F_ISSET(chunkcache, WT_CHUNK_CACHE_FLUSHED_DATA_INSERTION))
+        return (0);
+
+    WT_RET(__wt_open(session, local_name, WT_FS_OPEN_FILE_TYPE_DATA, WT_FS_OPEN_READONLY, &fh));
+    WT_ERR(__wt_filesize(session, fh, &size));
+
+    while (already_read < size) {
+        bucket_id =
+          __chunkcache_tmp_hash(chunkcache, &hash_id, sp_obj_name, objectid, already_read);
+
+        __wt_spin_lock(session, WT_BUCKET_LOCK(chunkcache, bucket_id));
+        ret =
+          __chunkcache_insert(session, (wt_off_t)already_read, size, &hash_id, bucket_id, &chunk);
```

Comment:
we don't need this cast on wt_off_t on already_read here either.

Reply:
updated in [af44def](https://github.com/wiredtiger/wiredtiger/pull/9677/commits/af44def3208cc8d59e29c45b05509a738ab79f1d)